utils/python/CIME/build.py

```diff
@@ -94,8 +94,9 @@ def build_model(build_threaded, exeroot, clm_config_opts, incroot, complist,
 
     config_dir = os.path.join(cimeroot, "driver_cpl", "cime_config")
     f = open(file_build, "w")
+    bldroot = os.path.join(exeroot, "cpl", "obj")
     stat = run_cmd("%s/buildexe %s %s %s" %
-                   (config_dir, caseroot, bldroot, libroot),
+                   (config_dir, caseroot, libroot, bldroot),
```

Comment:
In buildlib.py the positional arguments are input in the order
caseroot, bldroot, libroot but then returned in the order caseroot, libroot, bldroot
I'll fix this, but it'll involve changes in several files I think.